src/metrics/loc.rs

```diff
@@ -500,9 +500,55 @@ impl Loc for CppCode {
     }
 }
 
+impl fmt::Display for Java {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+impl Loc for JavaCode {
+    fn compute(node: &Node, stats: &mut Stats, is_func_space: bool, is_unit: bool) {
+        use Java::*;
+
+        let (start, end) = init(node, stats, is_func_space, is_unit);
+        let kind_id: Java = node.object().kind_id().into();
+        println!("{}", kind_id);
+        match kind_id {
+            Program => {}
+            Comment => {
+                add_cloc_lines(stats, start, end);
+            }
+            AssertStatement
+            | BreakStatement
+            | ContinueStatement
+            | DoStatement
+            | EnhancedForStatement
+            | ExpressionStatement
+            | IfStatement
+            | LocalVariableDeclaration
+            | ReturnStatement
+            | SwitchStatement
+            | ThrowStatement
+            | TryStatement
+            | WhileStatement => {
+                stats.logical_lines += 1;
+                println!("+1");
+            }
+            ForStatement => {
+                // handle binary check + increment
+                stats.logical_lines += 2;
```

Comment:
mmh... on wikipedia:
https://en.wikipedia.org/wiki/Source_lines_of_code
we have 1 for the `for` statement.
Tbh, from our pov, we don't care that much: we use these numbers in some machine learning stuff so I guess it doesn't make any difference at the end.
So I think we must be consistent in the way to compute those numbers across the different languages.
@marco-c, @Luni-4 do we have, in the scientific literature, some clear definitions ?

Reply:
@Luni-4 Thanks. A good suggestion to link in the code. I will add that.
I did read the article when you linked to it in [this discussion](https://github.com/mozilla/rust-code-analysis/issues/359#issuecomment-952631456) and we agreed then that the classic `for` loop had a LLOC of 3, in the same comment.

From that Oracle article:

> Assignment expressions
> Any use of ++ or --
> Method invocations
> Object creation expressions

Not counting the binary expression but counting the `for` loop itself (like is done with `while` and `do...while`), still puts us at a count of 3 if we include the `++` as the article suggests. This all depends if we decide that what is inside a `for` loop is subject to those rules.

[This article from Oracle](https://docs.oracle.com/javase/tutorial/java/nutsandbolts/for.html) refers to the initializer, condition, and incrementer as *expressions*. So since it is specific to the case we are talking about, my suggestion is we do as @Luni-4 suggests and make any `for` count as ` LLOC. That simplifies the code AND seems to be the most straightforward way to interpret those articles taken together.

So. `for` = 1 LLOC. Agreed?

Reply:
Since Java considers a `for` as a single statement and its internals as expressions, I agree with the `for` = 1 LLOC line. Let's link both the articles in the code as comments though, so we know we have decided one way instead of the other

Reply:
yep, agreed too.